### PR TITLE
[Gen 2] Add missing page for Switching Auth Flows

### PR DESCRIPTION
**Pull request by salihgueler**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [x] Swift
- [x] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by katiegoines — DISMISSED** [review]

**Review by renebrandel — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+
+## CUSTOM_AUTH flow
+
+Amazon Cognito User Pools supports customizing the authentication flow to enable custom challenge types, in addition to a password in order to verify the identity of users. The custom authentication flow is a series of challenge and response cycles that can be customized to meet different requirements. These challenge types may include CAPTCHAs or dynamic challenge questions.
+
+To define your challenges for custom authentication flow, you need to implement three Lambda triggers for Amazon Cognito.
+
+The flow is initiated by calling `signIn` with `AuthSignInOptions` configured with `AuthFlowType.customAuthWithSrp` OR `AuthFlowType.customAuthWithoutSrp`.
+
+Follow the instructions in [Custom Auth Sign In](/gen1/[platform]/build-a-backend/auth/sign-in-custom-flow/) to learn about how to integrate custom authentication flow in your application with the Auth APIs.
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx:349]: Should this be a gen1 link? I'm not to sure the page `sign-in-custom-flow` exists on either gen1 or gen2 pages

File: src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+
+## CUSTOM_AUTH flow
+
+Amazon Cognito User Pools supports customizing the authentication flow to enable custom challenge types, in addition to a password in order to verify the identity of users. The custom authentication flow is a series of challenge and response cycles that can be customized to meet different requirements. These challenge types may include CAPTCHAs or dynamic challenge questions.
+
+To define your challenges for custom authentication flow, you need to implement three Lambda triggers for Amazon Cognito.
+
+The flow is initiated by calling `signIn` with `AWSCognitoAuthSignInOptions` configured with `AuthFlowType.CUSTOM_AUTH_WITH_SRP` OR `AuthFlowType.CUSTOM_AUTH_WITHOUT_SRP`.
+
+Follow the instructions in [Custom Auth Sign In](/gen1/[platform]/build-a-backend/auth/sign-in-custom-flow/) to learn about how to integrate custom authentication flow in your application with the Auth APIs.
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx:484]: Should this be a gen1 link? Not sure if `sign-in-custom-flow` exists in either gen1 or gen2 pages

File: src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx (hunk 1 above)
Comment (salihgueler) [reply src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx]: It exists in gen2 at another branch.

File: src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx (hunk 2 above)
Comment (salihgueler) [reply src/pages/[platform]/build-a-backend/auth/switching-auth-flows/index.mdx]: It exists in gen2 at another branch.

**Review by renebrandel — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**